misc-scripts/xref_mapping/XrefParser/MGI_Desc_Parser.pm

```diff
@@ -18,135 +16,126 @@ limitations under the License.
-
   if ( !defined $mgi_io ) {
-    croak "ERROR: Could not open $file\n";
+    confess "Could not open $file\n";
   }
 
+  my $input_file = Text::CSV->new({
+    sep_char           => "\t",
+    quote_char         => undef,
+    escape_char        => undef,
+    strict             => 1,
+  }) or confess "Cannot use file $file: " . Text::CSV->error_diag();
+
   my $xref_count = 0;
   my $syn_count  = 0;
   my %acc_to_xref;
 
-
-    my $header = $mgi_io->getline(); #discard header line
-
-    chomp($header);
-    # crude emergency check for potentially altered file format.
-    my $header_template = qq(MGI Accession ID\tChr\tcM Position\tgenome coordinate start\tgenome coordinate end\tstrand\tMarker Symbol\tStatus\tMarker Name\tMarker Type\tFeature Type\tMarker Synonyms (pipe-separated));
-    if ($header ne $header_template) {die "File header has altered from format parser expects. Check MGI "};
-      
-    while ( my $line = $mgi_io->getline() ) {
-
-        chomp($line);
-        my ($accession, $chromosome, $position, $start, $end, $strand,$label, 
-            $status, $marker, $marker_type, $feature_type, $synonym_field) = split(/\t/,$line);
-            
-        $position =~ s/^\s+// if ($position);
-
-        my @synonyms = split(/\|/,$synonym_field) if ($synonym_field);
-        
-	
-        my $desc;
-	if ($marker) {
-	    $desc = $marker;
-	}
-        
-        $acc_to_xref{$accession} = $self->add_xref({ acc        => $accession,
-    	                           		             label      => $label,
-    					                             desc       => $desc,
-    					                             source_id  => $source_id,
-    					                             species_id => $species_id,
-                                                                     dbi        => $dbi,
-    					                             info_type  => "MISC"} );
-        if($verbose and !$desc){
-    	   print "$accession has no description\n";
-        }
-        $xref_count++;
-            
-        if(defined($acc_to_xref{$accession})){
-           
-            foreach my $syn (@synonyms) {
-                $self->add_synonym($acc_to_xref{$accession}, $syn, $dbi);
-                $syn_count++;
-            }
-            
-        }
-        
-    }
-      
-    $mgi_io->close();
-      
-    print $xref_count." MGI Description Xrefs added\n" if($verbose);
-    print $syn_count." synonyms added\n" if($verbose);
-
-  # expected columns
-  my @expected_columns =
-    qw(accession chromosome position start end strand label status marker marker_type feature_type synonym_field);
-
   # read header
-  my $header = $input_file->getline($mgi_io);
-  if ( scalar @{$header} != scalar @expected_columns ) {
-    croak "input file $file has an incorrect number of columns";
+  my @header_fields = $input_file->header($mgi_io);
+  if ( scalar @header_fields != $EXPECTED_NUMBER_OF_COLUMNS ) {
+    confess "Header of input file $file has an incorrect number of columns";
```

Comment:
As with #418, I think a check of the header fields can prevent future headaches if they change the format.

Reply:
Done.